packages/igx-templates/igx-ts/autocomplete/autocomplete-extended/files/src/app/__path__/__filePrefix__.component.ts

```diff
@@ -1,28 +1,35 @@
 import { Component, Pipe, PipeTransform, ViewChild } from '@angular/core';
-import { townsExtended } from './towns-data-extended';
-import { IgxToastComponent, IgxToastPosition } from '<%=igxPackage%>';
+import { townsExtended, Region } from './towns-data-extended';
+import { IgxToastComponent, IgxToastPosition, ISelectionEventArgs } from '<%=igxPackage%>';
 
 @Component({
     selector: 'app-<%=filePrefix%>',
     templateUrl: './<%=filePrefix%>.component.html',
     styleUrls: ['./<%=filePrefix%>.component.scss']
 })
 export class <%=ClassName%>Component {
-    public extendedTowns: any;
-    public townSelected: string;
-    public postalCode: string;
+    public regions!: Region[];
+    public townSelected!: string;
+    public postalCode!: number | 'none';
     public messagePosition = IgxToastPosition.Middle;
 
     @ViewChild(IgxToastComponent, { static: true })
-    public toast: IgxToastComponent;
+    public toast!: IgxToastComponent;
 
     constructor() {
-        this.extendedTowns = townsExtended;
+        this.regions = townsExtended;
     }
 
-    public getPostalCode(event) {
-        const targetRegion = this.extendedTowns.regions.filter((r) => r.name === event.newSelection.group.label)[0];
-        this.postalCode = targetRegion.towns.filter(t => t.name === event.newSelection.value)[0].postalCode;
+    public getPostalCode(e: ISelectionEventArgs) {
+
+        // Flatten the structure of the regions array
+        const arr = this.regions;
+        const allTowns = arr.map(region =>
+          region.towns.map(town => ({name: town.name, postalCode: town.postalCode}))).flat();
```

Comment:
Is `{name: town.name, postalCode: town.postalCode}` not the same as just the town object? And if so this is really `town => town` so no transform at all, and the same can be accomplished with 
```ts
townsExtended.map(region => region.towns).flat()
```
Also no point in the `arr` variable or the comment above it if the above suggestion is applies since it's fairly obvious what it does :)